### ISSUE TITLE
Issue when re-register Cassandra Source connector

### ISSUE THREAD

**Issue by CodeSmell** [Landoop/stream-reactor#364]
Using Kafka 1.0
Cassandra 3.11.1
Cassandra Source connector 1.0

Registered a connector (using curl POST)
Removed connector (using curl DELETE)
Attempted to re-register connector (using curl POST)

Getting this 
```
[2018-01-18 14:13:11,157] ERROR Failed to start task idocs-packs-0 (org.apache.kafka.connect.runtime.Worker:408)
java.lang.IllegalArgumentException: A metric named 'MetricName [name=offset-commit-max-time-ms, group=connector-task-metrics, description=The maximum time in milliseconds taken by this task to commit offsets., tags={connector=idocs-packs, task=0}]' already exists, can't register another one.
	at org.apache.kafka.common.metrics.Metrics.registerMetric(Metrics.java:532)
	at org.apache.kafka.common.metrics.Sensor.add(Sensor.java:256)
	at org.apache.kafka.common.metrics.Sensor.add(Sensor.java:241)
	at org.apache.kafka.connect.runtime.WorkerTask$TaskMetricsGroup.<init>(WorkerTask.java:328)
	at org.apache.kafka.connect.runtime.WorkerTask.<init>(WorkerTask.java:69)
	at org.apache.kafka.connect.runtime.WorkerSourceTask.<init>(WorkerSourceTask.java:100)
	at org.apache.kafka.connect.runtime.Worker.buildWorkerTask(Worker.java:445)
	at org.apache.kafka.connect.runtime.Worker.startTask(Worker.java:404)
	at org.apache.kafka.connect.runtime.distributed.DistributedHerder.startTask(DistributedHerder.java:852)
	at org.apache.kafka.connect.runtime.distributed.DistributedHerder.access$1600(DistributedHerder.java:108)
	at org.apache.kafka.connect.runtime.distributed.DistributedHerder$13.call(DistributedHerder.java:866)
	at org.apache.kafka.connect.runtime.distributed.DistributedHerder$13.call(DistributedHerder.java:862)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

looks like it is related to this... [Confluent issue 86](https://github.com/confluentinc/common/issues/86)

current work around = restart Kafka Connect

**Comment by stheppi** [comment]
This is far from ideal.
We will look to make sure stop somehow closes the poll

**Comment by stheppi** [comment]
Is a breaking change 1.0 and is the Connect framework responsiblity for freeing the resources it creates.The task is eventually killed by the framework , so the framework should also take care of cleaning up what it creates.

**Comment by CodeSmell** [comment]
reading some more and looks like it will be fixed in 1.0.1 or 1.1 of Kafka but these have not been released just yet. once they are released, can try it out again.

**Comment by iMajna** [comment]
Think you can close this issuse because its not connectors fault. Using this connectors for quite a good time and first time this issue appeared was after upgrade to Kafka1.0. and CP4.0.

If you take a look Kafkas jira on [https://issues.apache.org/jira/browse/KAFKA-6252](url) you can see the same @stheppi mentioned above.
